pkg/datapath/loader/loader.go

```diff
@@ -135,15 +170,35 @@ func nullifyStringSubstitutions(strings map[string]string) map[string]string {
 	return nullStrings
 }
 
+func (l *loader) bpfMasqAddrs(ifName string) (masq4, masq6 netip.Addr) {
+	if l.cfg.DeriveMasqIPAddrFromDevice != "" {
+		ifName = l.cfg.DeriveMasqIPAddrFromDevice
+	}
+
+	iter, _ := l.nodeAddrs.Get(
+		l.db.ReadTxn(),
+		tables.NodeAddressDeviceNameIndex.Query(ifName),
+	)
+	for addr, _, ok := iter.Next(); ok; addr, _, ok = iter.Next() {
```

Comment:
Dummy q, but where does the logic of `firstGlobal()` live now?

Reply:
It's still in `pkg/node/address_linux.go` as it's used by `node.SetDefaultPrefix`. It'll go away once that code is refactored.

Reply:
> now that I compare again, it might be I'll need to also include the IsPublicAddr logic for picking Primary

That's why I was asking the question :-)

Reply:
Addressed in https://github.com/cilium/cilium/pull/30579/.